src/main/java/com/ncc/aif/ValidateAIF.java

```diff
@@ -63,75 +75,304 @@ private static void loadModel(Model model, CharSource ontologySource) {
+                fileList.add(args[i++]);
+                numArgs++;
+            }
+        }
+        return numArgs; // Can't use fileList.size() just in case use specified same file twice
+    }
+
+    // Process command-line arguments, returning whether or not there were any errors.
+    private static boolean processArgs(String[] args, Set<String>flags, Set<String>domainOntologies,
+                                       Set<String> validationFiles, Set<String> validationDirs) {
+        boolean dashD = false;
+        boolean dashF = false;
+        for (int i = 0; i < args.length; i++) {
+            final String arg = args[i];
+            final String strippedArg = arg.trim();
+            switch (strippedArg) {
+                case "-h":
+                case "--help" :
+                    return false;
+                case "--ldc" :
+                    flags.add("LDC");
+                    break;
+                case "--program" :
+                    flags.add("PROGRAM");
+                    break;
+                case "--both" :
+                    flags.add("LDC");
+                    flags.add("PROGRAM");
+                    break;
+                case "--nist" :
+                    flags.add("NIST");
+                    break;
+                case "--ont" :
+                    int numFiles = processFiles(args, i, domainOntologies);
+                    if (numFiles == 0) {
+                        System.err.println("ERROR: --ont requires at least one ontology file to be specified.");
+                        return false;
+                    }
+                    i += numFiles;
+                    break;
+                case "-f" :
+                    if (dashD) {
+                        System.err.println("ERROR: Please specify either -d or -f, but not both.");
+                        return false;
+                    }
+                    i += processFiles(args, i, validationFiles);
+                    dashF = true;
+                    break;
+                case "-d" :
+                    if (dashF) {
+                        System.err.println("ERROR: Please specify either -d or -f, but not both.");
+                        return false;
+                    }
+                    if (!args[i + 1].startsWith("-")) {
+                        validationDirs.add(args[++i]);
+                        /* NOTE: if we choose to support validating files in N directories, change the above to:
+                         *   i += processFiles(args, i, validationDirs);
+                         */
+                    }
+                    dashD = true;
+                    break;
+                default:
+                    System.err.println("Ignoring unknown argument: " + arg);
+            }
+        }
+
+        if (!flags.contains("LDC") && !flags.contains("PROGRAM") && domainOntologies.isEmpty()) {
+            System.err.println("ERROR: Must validate against at least one domain ontology.");
```

Comment:
Is there a reason we are forcing the user to explicitly specify their ontology? If none are provided, couldn't we default to the program ontology?

Reply:
I personally prefer requiring them to be explicit since we have handy switches for LDC and Program, it matches the programmatic API, and that's what we came up with in that discussion in your office last week.  But if moving forward we know that for the most part people will be using the program ontology, then I'm not opposed to a default either.  It feels to me, though, like something to change on a future pass after the performers have taken a look at the proposed validator.

Reply:
I also prefer users being specific.  This matches the use case discussed at the Jan 2019 PI meeting.